webview-ui/src/ConnectAcrToCluster/ConnectAcrToCluster.tsx

```diff
@@ -0,0 +1,280 @@
+import { useEffect } from "react";
+import {
+    Acr,
+    Cluster,
+    InitialState,
+    Subscription,
+} from "../../../src/webview-contract/webviewDefinitions/connectAcrToCluster";
+import { Lazy, isLoaded, map as lazyMap } from "../utilities/lazy";
+import { useStateManagement } from "../utilities/state";
+import styles from "./ConnectAcrToCluster.module.css";
+import {
+    EventHandlerFunc,
+    ensureAcrRoleAssignmentLoaded,
+    ensureAcrsLoaded,
+    ensureClustersLoaded,
+    ensureSubscriptionsLoaded,
+} from "./state/dataLoading";
+import { ConnectAcrToClusterState, stateUpdater, vscode } from "./state/state";
+import { distinct } from "../utilities/array";
+import { ResourceSelector } from "../components/ResourceSelector";
+import { VSCodeButton } from "@vscode/webview-ui-toolkit/react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { IconDefinition, faCheckCircle, faClock, faSave, faTrash } from "@fortawesome/free-solid-svg-icons";
+import { AcrRoleState } from "./state/stateTypes";
+
+export function ConnectAcrToCluster(initialState: InitialState) {
+    const { state, eventHandlers } = useStateManagement(stateUpdater, initialState, vscode);
+
+    const updates: EventHandlerFunc[] = [];
+    const {
+        lazySubscriptions,
+        lazyAcrResourceGroups,
+        lazyClusterResourceGroups,
+        lazyAcrs,
+        lazyClusters,
+        lazyAcrRoleState,
+    } = prepareData(state, updates);
+    useEffect(() => {
+        updates.map((fn) => fn(eventHandlers));
+    });
+
+    function getAcrAuthorizationActionItemProps(): ActionItemProps {
+        const createAction = makeFixAction(faSave, "Authorize", null);
```

Comment:
passing action as null here was confusing for me , but then realized we need sub, rg etc.. other properties selected so that you can enabled authorize and deauthorize buttons. I would suggest getting UX validations for these terms from Quentin or Ivelisse. But can you please add a comment why action was selected later for other understanding. May be is it possible to simple this method ?

```
createAction.canPerformAction = !isDone;
        createAction.action = () => handleCreateAcrRoleAssignment(acr, cluster);
        deleteAction.canPerformAction = isDone;
        deleteAction.action = () => handleDeleteAcrRoleAssignment(acr, cluster);

```

Reply:
Commented.

@qpetraroia: please share if you have some better names here. I went with 'Authorize'/'Deauthorize' because that's what 'connect' really means: you add role assignments to _authorize_ the cluster identity to pull images from the ACR.

My main intent is to get something working that can be iterated on. This has been the spirit of all work so far, rather than aiming for 'perfect'. With all work, we have to make a lot of best guesses, and those guesses won't always be right.